src/scheduler/job_info.c

```diff
@@ -4189,11 +4175,13 @@ formula_evaluate(char *formula, resource_resv *resresv, resource_req *resreq)
 	}
 
 	/* special cases */
-	sprintf(buf, "\'%s\':%ld,\'%s\':%d,\'%s\':%d,\'%s\':%lf, \'%s\':%d}",
+	sprintf(buf, "\'%s\':%ld,\'%s\':%d,\'%s\':%d,\'%s\':%f, \'%s\': %f, \'%s\': %f, \'%s\':%d}",
 		FORMULA_ELIGIBLE_TIME, resresv->job->eligible_time,
 		FORMULA_QUEUE_PRIO, resresv->job->queue->priority,
 		FORMULA_JOB_PRIO, resresv->job->priority,
-		FORMULA_FSPERC, resresv->job->ginfo->percentage,
+		FORMULA_FSPERC, resresv->job->ginfo->tree_percentage,
```

Comment:
I made sure the new fairshare_factor keyword won't divide by zero.  If it would, I use zero for the value instead.  I added text to the EDD to warn admins to not have entities with zero shares if they use the formula.  SInce the formula is opaque to us, there isn't really any more I can do.  The formula evaluation is in a try/except block, so nothing bad happens to the scheduler.  If a formula throws an exception, zero is used for the evaluated value.